specs/Picker-API_spec.md

```diff
@@ -0,0 +1,116 @@
+# Background
+
+The current FilePicker API for UWP has several limitations. Some of the top limitations 
+are as described below:
+
+**1. Unable to pick files and folders in a single instance of the Picker dialog**
+
+   The Picker dialog allows developers to specify picker mode to be single file or multiple
+   files. It does not however, allow developers to specify a mode to pick both files and folders.
+   There has been feedback indicating that this behaviour is limiting, requiring developers
+   to build their own picker experience. 
+
+**2. Unable to pick multiple folders.**
+
+   The Picker dialog (FolderPicker) does not allow users to pick multiple folders. The developer
+   experience currently is to select a single folder and repeat the dialog for more folders. This 
+   is a less than ideal experience for application developers.
+
+The Reunion picker dialog is going to address the above issues. This Reunion API will be based upon the below
+APIs in the Windows SDK. The Reunion Picker API will remove the deprecated APIs and add additional methods
+for the functionality: File and folder picker, Multiple folder picker.
+
+- Windows.Storage.Pickers.FileOpenPicker (https://docs.microsoft.com/en-us/uwp/api/Windows.Storage.Pickers.FileOpenPicker)
+- Windows.Storage.Pickers.FileSavePicker (https://docs.microsoft.com/en-us/uwp/api/windows.storage.pickers.filesavepicker)
+- Windows.Storage.Pickers.FolderPicker (https://docs.microsoft.com/en-us/uwp/api/windows.storage.pickers.folderpicker) 
+
+# Description
+
+The goal of this API is to address the gaps in the existing File/Folder picker APIs. The API surface is similar to the existing
+Picker APIs in Windows SDK with additional methods to support File + Folder picker and multiple folder picker. The API removes the
+deprecated methods from Windows SDK.
+
+# Examples
```

Comment:
A single file/folder open picker API would be optimal, save picker can be own its own. Here is my take on how the API usage would like. It also have some naming suggestions. Basically have a single open picker and provide options to configure it the way the app developer would like to have. Feedback welcome!

enum PickerMode
{
    Any = 0,
    File,
    Folder
}

## Pick files and folders

var picker = new StorageItemPicker();
picker.StartLocation = startLocation;
picker.PickMultiple = true;
picker.PickerMode = PickerMode.Any;
picker.FileFilters.add("*.jpg");
picker.FileFilters.add("*.png");
IReadOnlyList<StorageItem> storageItems = await picker.PickStorageItemsAsync();

## Pick single file

var picker = new StorageItemPicker();
picker.StartLocation = startLocation;
picker.PickMultiple = false;
picker.PickerMode = PickerMode.File;
IReadOnlyList<StorageItem> storageItems = await picker.PickStorageItemsAsync();

## Pick multiple files

var picker = new StorageItemPicker();
picker.StartLocation = startLocation;
picker.PickMultiple = true;
picker.PickerMode = PickerMode.File;
picker.FileFilters.add("*.jpg");
picker.FileFilters.add("*.png");
IReadOnlyList<StorageItem> storageItems = await picker.PickStorageItemsAsync();

## Pick single folder

var picker = new StorageItemPicker();
picker.StartLocation = startLocation;
picker.PickMultiple = false;
picker.PickerMode = PickerMode.Folder;
IReadOnlyList<StorageItem> storageItems = await picker.PickStorageItemsAsync();

## Pick multiple folders

var picker = new StorageItemPicker();
picker.StartLocation = startLocation;
picker.PickMultiple = true;
picker.PickerMode = PickerMode.Folder;
IReadOnlyList<StorageItem> storageItems = await picker.PickStorageItemsAsync();

## Save a file
var picker = new FileSavePicker();
picker.StartLocation = startLocation;
StorageFile savedFile = await picker.SaveFileAsync();